File: leetcode/1726.同积元组.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = defaultdict(int)
        for i in range(n):
            for j in range(i + 1, n):
                cnt[nums[i] * nums[j]] += 1
        ans = 0
        for k, v in cnt.items():
            ans += v * (v - 1) * 4
        return ans 

File: leetcode/1726.同积元组.go
func tupleSameProduct(nums []int) int {
    cnt := make(map[int]int)
    for i := 0; i < len(nums); i++ {
        for j := i + 1; j < len(nums); j++ {
            cnt[nums[i] * nums[j]] += 1
        }
    }
    ans := 0
    for _, v := range cnt {
        ans += v * (v - 1) * 4
    }    
    return ans
}